pageserver/benches/bench_walredo.rs

```diff
@@ -166,12 +157,46 @@ macro_rules! lsn {
     }};
 }
 
-/// Short payload, 1132 bytes.
-// pg_records are copypasted from log, where they are put with Debug impl of Bytes, which uses \0
-// for null bytes.
-#[allow(clippy::octal_escapes)]
-fn short() -> Request {
-    Request {
+/// Simple wrapper around `WalRedoManager::request_redo`.
+///
+/// In benchmarks this is cloned around.
+#[derive(Clone)]
+struct Request {
+    key: Key,
+    lsn: Lsn,
+    base_img: Option<(Lsn, Bytes)>,
+    records: Vec<(Lsn, NeonWalRecord)>,
+    pg_version: u32,
+}
+
+impl Request {
+    async fn execute(&self, manager: &PostgresRedoManager) -> anyhow::Result<Bytes> {
+        let Request {
+            key,
+            lsn,
+            base_img,
+            records,
+            pg_version,
+        } = self;
+
+        // TODO: avoid these clones
+        manager
+            .request_redo(*key, *lsn, base_img.clone(), records.clone(), *pg_version)
+            .await
```

Comment:
We could make both of these requests into OnceLocks and init them only once, then just send them over as borrowed? Or are we cloning a reference here?

Reply:
The costly copy is the `records.clone()`, it clones the `Vec` although I believe each element is Bytes::clone()d and hence it's cheap (?)

I don't intend to resovle this TODO as part of this PR.
I think the way forward is to change the `request_redo` interface.
We can do it in a future PR and use this benchmark to compare the results.

Reply:
I did not remember that the ownership is moved to the fn for the vec, I was expecting otherwise. I cannot see why right away...

> Bytes::clone()d and hence it's cheap

The NeonWalRecord might be still quite large, but I don't think there's anything to be done about that.